src/org/labkey/mobileappstudy/surveydesign/ServiceSurveyDesignProvider.java

```diff
@@ -105,7 +106,14 @@ private static String getServiceUrl(Container container)
     {
         Module module = ModuleLoader.getInstance().getModule(MobileAppStudyModule.NAME);
         String value = module.getModuleProperties().get(MobileAppStudyModule.METADATA_SERVICE_BASE_URL).getEffectiveValue(container);
-        return value == null ? null : value.trim();
+        if (StringUtils.isBlank(value))
+            return null;
+        value = value.trim();
```

Comment:
Could wrap initial assignment with StringUtils.trimToNull() and then simple null check.